mat64/dense_arithmetic.go

```diff
@@ -351,6 +351,7 @@ func (m *Dense) Dot(b Matrix) float64 {
 }
 
 // Mul takes the matrix product of a and b, placing the result in the receiver.
+// If a or b are also the receiver, a workspace will be allocated.
```

Comment:
I haven't read the csparse or sparseBLAS documentation, so that may be the problem. When someone says "workspace", I think of something of bigger importance than allocating a []float64. For example, a workspace in Matlab is all of the variables that you have in memory. I've heard it used in python as a synonym for "environment", where it refers to the importable packages and where they live. Perhaps falsely, but I also associate the term with what's talked about in fftw and the CUDA documentation, where it is a memory allocation issue, but there are also a whole bunch of parameters you can tweak to get the behavior you want. In go, memory allocation is a relatively small deal with GC, and the important parts beyond the memory (size, shape, etc.) are handled within the function, not by the user. It matters for performance (I agree that the documentation is good), but 'workspace' to me implies something I need to care after and configure, while here we're just letting the user know that a significant memory allocation may happen.